packages/plugin-breadcrumbs-console/src/index.ts

```diff
@@ -46,15 +46,28 @@ function consoleBreadcrumbsPlugin(options?: { [key: string]: any }) {
 }
 
 function serializeValue(value: any, method: string) {
-  if (typeof value === "string") {
-    return value
-  } else {
-    try {
-      return JSON.stringify(value)
-    } catch (error) {
-      console.error(`Could not serialize "console.${method}" to String.`, error)
-      return "[Value could not be serialized]"
+  switch (typeof value) {
+    case "string":
+      return value
+    case "undefined":
+      return "undefined"
+    default:
+      return JSON.stringify(value, circularReplacer())
+  }
+}
+
+function circularReplacer() {
+  const seen = new WeakMap()
```

Comment:
[WeakMap](http://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap) isn't available on some older browsers, like Internet Explorer < 11. We  say we still support down to IE9. I've suggested dropping it in https://github.com/appsignal/integration-guide/issues/53 (private link), but as we still support it we need a polyfill or alternative implementation.

Previously, for the Promise polyfill, we've not shipped it with the library. We can also say it's up to the user to add it? https://docs.appsignal.com/front-end/#supported-browsers

Reply:
Good catch! I think I'll change it so that it does something like [what cycle.js did before it used WeakMap.](https://github.com/douglascrockford/JSON-js/commit/e25ba46913160617935f837836016ea7ed2be08c)